pyccl/ccl_utils.i

```diff
@@ -122,4 +136,48 @@ void get_spline2d_arrays(gsl_spline2d *spline2d,
   memcpy(zarr, spline2d->zarr, sizeof(double)*x_size*y_size);
 }
 
+void get_spline3d_arrays(gsl_spline2d **spline2d,
+                         int x_size, double* xarr,
+                         int y_size, double* yarr,
+                         int t_size, double* tarr,
+                         int na,
+                         int *status)
+{
+  // check for inconsistencies //
+  if(spline2d == NULL) {
+    *status = CCL_ERROR_MEMORY;
+    return;
+  }
+  // check only the first array element; the others must be consistent //
+  if(x_size != spline2d[0]->interp_object.xsize) {
+    *status = CCL_ERROR_INCONSISTENT;
+    return;
+  }
+  if(y_size != spline2d[0]->interp_object.ysize) {
+    *status = CCL_ERROR_INCONSISTENT;
+    return;
+  }
+
+  for (int ia = 0; ia < na; ia++) {
+    for (int ik = 0; ik < x_size*y_size; ik++) {
+      tarr[ia*x_size*y_size + ik] = spline2d[ia]->zarr[ik];
+    }
+  }
+  // no need to do this for every scale factor //
+  memcpy(xarr, spline2d[0]->xarr, sizeof(double)*x_size);
+  memcpy(yarr, spline2d[0]->yarr, sizeof(double)*y_size);
+}
+
+
+void get_array(double *arr,
```

Comment:
Is that just to copy the data? I thought python could already access raw arrays.

Reply:
Yes, just to copy the data. I can't access them as raw arrays:
```
In [10]: tsp.tsp.a_arr
Out[10]: <Swig Object of type 'double *' at 0x7f63880ae600>
```

Reply:
Ah, maybe I'm confusing it with `ctypes`.